From 820df42a87322eef69a8d02451caa705d53b7dda Mon Sep 17 00:00:00 2001
From: Michael Stahl <mstahl@redhat.com>
Date: Thu, 30 Jan 2014 12:12:00 +0100
Subject: [PATCH] stop linking to unneeded Xmu, Xi, Xext libraries

Apparently Xmu and Xi were needed when glewinfo was using glut, but that
dependency was removed 11 years ago in
031afdf03be1d123b7a442f64cfa28c6066c1490.
---
 auto/doc/build.html           | 2 +-
 config/Makefile.cygwin        | 2 +-
 config/Makefile.darwin        | 2 +-
 config/Makefile.darwin-ppc    | 2 +-
 config/Makefile.darwin-x86_64 | 2 +-
 config/Makefile.freebsd       | 2 +-
 config/Makefile.gnu           | 2 +-
 config/Makefile.kfreebsd      | 2 +-
 config/Makefile.linux         | 2 +-
 config/Makefile.netbsd        | 2 +-
 config/Makefile.openbsd       | 2 +-
 config/Makefile.solaris       | 2 +-
 config/Makefile.solaris-gcc   | 2 +-
 doc/build.html                | 2 +-
 14 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/config/Makefile.cygwin b/config/Makefile.cygwin
index b044273..8600fd0 100644
--- a/config/Makefile.cygwin
+++ b/config/Makefile.cygwin
@@ -6,7 +6,7 @@ LD := cc
 LN :=
 LDFLAGS.EXTRA =
 LIBDIR = $(GLEW_DEST)/lib
-LDFLAGS.GL = -lXmu -lXi -lGL -lXext -lX11
+LDFLAGS.GL = -lGL -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
 WARN = -Wall -W
diff --git a/config/Makefile.darwin b/config/Makefile.darwin
index 0c05ed1..bf34a0e 100644
--- a/config/Makefile.darwin
+++ b/config/Makefile.darwin
@@ -6,7 +6,7 @@ CFLAGS.EXTRA = -dynamic -fno-common
 LDFLAGS.EXTRA =
 ifneq (undefined, $(origin GLEW_APPLE_GLX))
 CFLAGS.EXTRA += -I/usr/X11R6/include -D'GLEW_APPLE_GLX'
-LDFLAGS.GL = -L/usr/X11R6/lib -lXmu -lXi -lGL -lXext -lX11
+LDFLAGS.GL = -L/usr/X11R6/lib -lGL -lX11
 else
 LDFLAGS.GL = -framework AGL -framework OpenGL
 endif
diff --git a/config/Makefile.darwin-ppc b/config/Makefile.darwin-ppc
index 2aaf66a..46c8b73 100644
--- a/config/Makefile.darwin-ppc
+++ b/config/Makefile.darwin-ppc
@@ -6,7 +6,7 @@ CFLAGS.EXTRA = -arch ppc -dynamic -fno-common
 LDFLAGS.EXTRA = -arch ppc
 ifneq (undefined, $(origin GLEW_APPLE_GLX))
 CFLAGS.EXTRA += -I/usr/X11R6/include -D'GLEW_APPLE_GLX'
-LDFLAGS.GL = -L/usr/X11R6/lib -lXmu -lXi -lGL -lXext -lX11
+LDFLAGS.GL = -L/usr/X11R6/lib -lGL -lX11
 else
 LDFLAGS.GL = -framework AGL -framework OpenGL
 endif
diff --git a/config/Makefile.darwin-x86_64 b/config/Makefile.darwin-x86_64
index 83e5c47..e6eb050 100644
--- a/config/Makefile.darwin-x86_64
+++ b/config/Makefile.darwin-x86_64
@@ -6,7 +6,7 @@ CFLAGS.EXTRA = -arch x86_64 -dynamic -fno-common
 LDFLAGS.EXTRA = -arch x86_64
 ifneq (undefined, $(origin GLEW_APPLE_GLX))
 CFLAGS.EXTRA += -I/usr/X11R6/include -D'GLEW_APPLE_GLX'
-LDFLAGS.GL = -L/usr/X11R6/lib -lXmu -lXi -lGL -lXext -lX11
+LDFLAGS.GL = -L/usr/X11R6/lib -lGL -lX11
 else
 LDFLAGS.GL = -framework AGL -framework OpenGL
 endif
diff --git a/config/Makefile.freebsd b/config/Makefile.freebsd
index ab9d9d9..bb66fcc 100644
--- a/config/Makefile.freebsd
+++ b/config/Makefile.freebsd
@@ -2,7 +2,7 @@ NAME = $(GLEW_NAME)
 CC = cc
 LD = ld
 LDFLAGS.EXTRA = -L/usr/X11R6/lib
-LDFLAGS.GL = -lXmu -lXi -lGL -lXext -lX11
+LDFLAGS.GL = -lGL -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
 CFLAGS.EXTRA += -I/usr/X11R6/include
diff --git a/config/Makefile.gnu b/config/Makefile.gnu
index 2398ab0..b7c58be 100644
--- a/config/Makefile.gnu
+++ b/config/Makefile.gnu
@@ -2,7 +2,7 @@ NAME = $(GLEW_NAME)
 CC = cc
 LD = cc
 LDFLAGS.EXTRA = -L/usr/X11R6/lib
-LDFLAGS.GL = -lXmu -lXi -lGL -lXext -lX11
+LDFLAGS.GL = -lGL -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
 NAME = GLEW
diff --git a/config/Makefile.kfreebsd b/config/Makefile.kfreebsd
index 9501091..faf1046 100644
--- a/config/Makefile.kfreebsd
+++ b/config/Makefile.kfreebsd
@@ -2,7 +2,7 @@ NAME = $(GLEW_NAME)
 CC = cc
 LD = cc
 LDFLAGS.EXTRA = -L/usr/X11R6/lib
-LDFLAGS.GL = -lXmu -lXi -lGL -lXext -lX11
+LDFLAGS.GL = -lGL -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
 NAME = GLEW
diff --git a/config/Makefile.linux b/config/Makefile.linux
index 234adc6..b460b4b 100644
--- a/config/Makefile.linux
+++ b/config/Makefile.linux
@@ -16,7 +16,7 @@ else
   LDFLAGS.EXTRA = -L/usr/X11R6/lib -L/usr/lib
   LIBDIR = $(GLEW_DEST)/lib
 endif
-LDFLAGS.GL = -lXmu -lXi -lGL -lXext -lX11
+LDFLAGS.GL = -lGL -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
 NAME = GLEW
diff --git a/config/Makefile.netbsd b/config/Makefile.netbsd
index 5f69cd6..6da47b1 100644
--- a/config/Makefile.netbsd
+++ b/config/Makefile.netbsd
@@ -2,7 +2,7 @@ NAME = $(GLEW_NAME)
 CC = cc
 LD = ld
 LDFLAGS.EXTRA = -L/usr/X11R7/lib -R /usr/X11R7/lib 
-LDFLAGS.GL = -lXmu -lXi -lGL -lXext -lX11
+LDFLAGS.GL = -lGL -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
 CFLAGS.EXTRA += -I/usr/X11R7/include -fPIC
diff --git a/config/Makefile.openbsd b/config/Makefile.openbsd
index 365ea9e..ade993e 100644
--- a/config/Makefile.openbsd
+++ b/config/Makefile.openbsd
@@ -2,7 +2,7 @@ NAME = $(GLEW_NAME)
 CC = cc
 LD = ld
 LDFLAGS.EXTRA = -L/usr/X11R6/lib
-LDFLAGS.GL = -lXmu -lXi -lGLU -lGL -lXext -lX11 -lm
+LDFLAGS.GL = -lGLU -lGL -lX11 -lm
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
 CFLAGS.EXTRA += -I/usr/X11R6/include
diff --git a/config/Makefile.solaris b/config/Makefile.solaris
index befba5a..fe78ca1 100644
--- a/config/Makefile.solaris
+++ b/config/Makefile.solaris
@@ -4,7 +4,7 @@ LD = ld
 CFLAGS.EXTRA = -I/usr/openwin/include
 LDFLAGS.SO = -G
 LDFLAGS.EXTRA = -L/usr/openwin/lib
-LDFLAGS.GL = -lXmu -lXi -lGL -lXext -lX11
+LDFLAGS.GL = -lGL -lX11
 NAME = GLEW
 BIN.SUFFIX =
 POPT = -xO2
diff --git a/config/Makefile.solaris-gcc b/config/Makefile.solaris-gcc
index c0373f4..e00a95b 100644
--- a/config/Makefile.solaris-gcc
+++ b/config/Makefile.solaris-gcc
@@ -4,7 +4,7 @@ LD = ld
 CFLAGS.EXTRA = -I/usr/openwin/include
 LDFLAGS.SO = -G
 LDFLAGS.EXTRA = -L/usr/openwin/lib
-LDFLAGS.GL = -lXmu -lXi -lGL -lXext -lX11
+LDFLAGS.GL = -lGL -lX11
 NAME = GLEW
 BIN.SUFFIX =
 POPT = -O2
-- 
1.8.3.1

